### PR TITLE
add mpoly-parametric file for Q(t)[x]

### PR DESCRIPTION
**Pull request by tthsqe12**

### CODE REVIEW

File: src/Rings/mpoly-parametric.jl [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+# if R2 is a singular ring, keep it that way since limited types can be factored
+function _add_variables(R2::MPolyRing, n::Int)
+   n += nvars(R2)
+   x = map(i->"x"*string(i), 1:n)
+   if isa(R2, Singular.PolyRing)
+      return Singular.PolynomialRing(base_ring(R2), x)[1]
+   else
+      return PolynomialRing(base_ring(R2), x)[1]
+   end
+end
```
Comment (thofma) [inline src/Rings/mpoly-parametric.jl:62]: Why not two methods without the `isa` shenanigans? Also the `PolynomialRing` calls should disable caching.

**Review by fieker — COMMENTED** [review]
what about a more general function to transform towers of MPoly/FracElem to Mpoly and back? Not just two steps?

**Comment by tthsqe12** [comment]
Ok. I can do towers of Poly/MPoly. Not sure if it is worth putting arbitrary Frac's in there.

**Comment by thofma** [comment]
There were still some things to fix ...

**Comment by tthsqe12** [comment]
What were the problems? Missing tests?

**Comment by thofma** [comment]
Sorry, I forgot to "submit" my review. Did it now.

**Comment by tthsqe12** [comment]
Ok. So you really don't like isa? I have more to add to this file, so I can fix these soon.

**Comment by thofma** [comment]
Yeah, if we can avoid it, we should avoid it.

**Pull request merged**